org.eclipse.xtext/src/org/eclipse/xtext/workspace/ISourceFolder.java

```diff
@@ -20,4 +32,28 @@
 	 * separator. It will never be null.
 	 */
 	URI getPath();
+
+	/**
+	 * @param uri
+	 *            to check
+	 * @return true iff the given {@link URI} is a sub element of the {@link URI} of this {@link ISourceFolder}.
+	 */
+	default boolean contains(URI uri) {
+		URI path = getPath();
+		return UriUtil.isPrefixOf(path, uri);
+	}
+
+	/** @return a list of all URIs that are passed to the acceptor of {@link #scan(URI, IAcceptor)} */
+	default List<URI> getAllResources() {
+		List<URI> uris = new LinkedList<>();
```

Comment:
Please use ArrayList instead. In general it is the better choice compared to LinkedList. If things are used as a queue, ArrayDeque is superior to LinkedList.